classes/class.memberorder.php

```diff
@@ -10,15 +411,192 @@ function __construct($id = NULL)
 			$this->setGateway(pmpro_getOption("gateway"));
 
 			//get data if an id was passed
-			if($id)
-			{
+			if( $id ) {
 				if(is_numeric($id))
-					return $this->getMemberOrderByID($id);
+					$morder = $this->getMemberOrderByID($id);
 				else
-					return $this->getMemberOrderByCode($id);
+					$morder = $this->getMemberOrderByCode($id);
+			} else {
+				$morder = $this->getEmptyMemberOrder();	//blank constructor
 			}
-			else
-				return $this->getEmptyMemberOrder();	//blank constructor
+
+			$this->original_status = $this->status;
+
+			return $morder;
+		}
+
+		/**
+		 * Get a specific order by ID or code
+		 *
+		 * @since TBA
+		 * 
+		 * @param mixed $id Specify an order ID or code that you want to retrieve
+		 *
+		 */
+		public static function get_order( $id = NULL ) {
+
+			return new MemberOrder( $id );
+
+		}
```

Comment:
This function should take an array of arguments just as the `get_orders()` method does, and should leverage that function to avoid duplicate logic.

For reference, see `get_subscription()` method from our subs table work and how it "forwards" the call to `get_subscriptions()`: https://github.com/strangerstudios/paid-memberships-pro/blob/951e7008b9a69e804d9471c171bdf85b39dc72a0/classes/class-pmpro-subscription.php#L262-L299